resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ResourceTree.java

```diff
@@ -345,27 +345,48 @@ private boolean internalDeleteFolder(IFolder folder, int flags, IProgressMonitor
 		if (!folder.exists())
 			return true;
 
-		// Don't delete contents if this is a linked resource
-		if (folder.isLinked()) {
-			deletedFolder(folder);
-			return true;
-		}
-
-		// If the folder doesn't exist on disk then update the tree and return.
 		IFileStore fileStore = localManager.getStore(folder);
-		if (!fileStore.fetchInfo().exists()) {
-			deletedFolder(folder);
-			return true;
-		}
+		if (ZipFileUtil.isOpenZipFile(fileStore)) {
+			try {
+				// folder is opened zip file
+				deletedFolder(folder);
+				// if the IResource.CLOSE_ZIP_FILE flag is set, the file should not be deleted
+				// in the file system.
+				if ((flags & (IResource.CLOSE_ZIP_FILE)) != 0) {
+					return true;
+				}
+				IFile file = folder.getParent().getFile(IPath.fromOSString(folder.getName()));
+				IFileStore parentStore = localManager.getStore(file);
+				parentStore.delete(EFS.NONE, Policy.subMonitorFor(monitor, Policy.totalWork / 4));
+				return true;
+			} catch (Exception e) {
```

Comment:
avoid e.printStackTrace();

Reply:
I improved the error handling. Thanks for the hint!